### PR TITLE
Clean up dependencies / devDependencies

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
* Updated dependencies to latest version 
    * Resolves https://github.com/cerner/generator-terra-module/issues/107
* Uplifted generator to use latest yeoman-generator conventions 
    * Resolves https://github.com/cerner/generator-terra-module/issues/92
* Updated to ESLint 5 and fixed lint issues 
    * Resolves https://github.com/cerner/generator-terra-module/issues/105 && Resolves https://github.com/cerner/generator-terra-module/issues/111
* Replaced gulp with npm scripts
* Removed [nsp as its going offline in 9/30/2018](https://blog.npmjs.org/post/175511531085/the-node-security-platform-service-is-shutting)

To test this change out:
* checkout this branch locally
* run npm install in this branch
* make a new sibiling directory to the `generator-terra-module` module directory  and `cd` into it
* run `yo ../generator-terra-module/generators/app` (this will use this branch's version of generator-terra-module)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,12 +23,12 @@
...
     "lodash.camelcase": "^4.3.0",
     "lodash.startcase": "^4.4.0",
     "lodash.upperfirst": "^4.3.1",
-    "yeoman-generator": "^0.22.0",
-    "yosay": "^1.0.2"
+    "yeoman-generator": "^3.1.1",
+    "yosay": "^2.0.2"
   },
   "devDependencies": {
     "eslint": "^2.1.0",
```
Comment (yuderekyu) [inline package.json]: want to update the yeoman test dependencies too? noticed `yeoman-assert` is on `3.1.1` and `yeoman-test` is on `2.0.0`

Comment (bjankord) [reply package.json]: Updated to latest release

File: package.json [hunk 2]
```diff
@@ -23,38 +23,36 @@
...
-    "gulp-mocha": "^2.0.0",
-    "gulp-nsp": "^2.1.0",
-    "gulp-plumber": "^1.0.0",
-    "slugg": "^1.0.0",
-    "yeoman-assert": "^2.0.0",
+    "eslint": "^5.0.0",
+    "eslint-config-terra": "^2.0.0",
+    "mocha": "^5.2.0",
+    "yeoman-assert": "^3.0.0",
     "yeoman-test": "^1.0.0"
```
Comment (yuderekyu) [inline package.json]: any reason why you didn't choose to update `yeoman-test`?

Comment (bjankord) [reply package.json]: Ah, I missed that 2.0.0 was available, I'll get that updated in this PR.

Comment (bjankord) [reply package.json]: Updated: https://github.com/cerner/generator-terra-module/pull/110/commits/fd9e1b154f01ea899452d526013866202efe4064

**Pull request merged**